launch/test/launch/substitutions/test_path_join_substitution.py

```diff
@@ -31,3 +31,7 @@ def test_path_join():
     path = ['path', ['to'], ['my_', TextSubstitution(text='file'), '.yaml']]
     sub = PathJoinSubstitution(path)
     assert sub.perform(context) == os.path.join('path', 'to', 'my_file.yaml')
+
+    sub = PathSubstitution('some') / 'path'
+    sub = sub / PathJoinSubstitution(['to', 'some', 'file'])
+    assert sub.perform(context) == os.path.join('some', 'path', 'to', 'some', 'file')
```

Comment:
Could you also test `/` with an `Iterable[Union[Text, Substitution]]`, since it's part of `SomeSubstitutionsType`? Not sure if this works, but it should:

```suggestion
    sub = PathSubstitution('some') / 'path'
    sub = sub / PathJoinSubstitution(['to', 'some', 'dir'])
    sub = sub / (TextSubstitution(text='my_model'), '.xacro')
    assert sub.perform(context) == os.path.join('some', 'path', 'to', 'some', 'dir', 'my_model.xacro')
```